plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/RestUpdateByQueryAction.java

```diff
@@ -0,0 +1,94 @@
+
+    @Inject
+    public RestUpdateByQueryAction(Settings settings, RestController controller, Client client,
+            IndicesQueriesRegistry indicesQueriesRegistry) {
+        super(settings, controller, client);
+        this.indicesQueriesRegistry = indicesQueriesRegistry;
+        controller.registerHandler(POST, "/{index}/_update_by_query", this);
+        controller.registerHandler(POST, "/{index}/{type}/_update_by_query", this);
+    }
+
+    @Override
+    protected void handleRequest(RestRequest request, RestChannel channel, Client client) throws Exception {
+        /*
+         * Passing the search request through UpdateByQueryRequest first allows
+         * it to set its own defaults which differ from SearchRequest's
+         * defaults. Then the parse can override them.
+         */
+        UpdateByQueryRequest internalRequest = new UpdateByQueryRequest(new SearchRequest());
+        int batchSize = internalRequest.source().source().size();
+        internalRequest.source().source().size(-1);
+        /*
+         * We can't send parseSearchRequest REST content that it doesn't support
+         * so we will have to remove the content that is valid in addition to
+         * what it supports from the content first. This is a temporary hack and
+         * should get better when SearchRequest has full ObjectParser support
+         * then we can delegate and stuff.
+         */
+        BytesReference bodyContent = null;
+        if (RestActions.hasBodyContent(request)) {
+            bodyContent = RestActions.getRestContent(request);
+            Tuple<XContentType, Map<String, Object>> body = XContentHelper.convertToMap(bodyContent, false);
+            boolean modified = false;
+            String conflicts = (String) body.v2().remove("conflicts");
+            if (conflicts != null) {
+                internalRequest.conflicts(conflicts);
+                modified = true;
+            }
+            @SuppressWarnings("unchecked")
+            Map<String, Object> script = (Map<String, Object>) body.v2().remove("script");
+            if (script != null) {
+                internalRequest.script(Script.parse(script, false, parseFieldMatcher));
+                modified = true;
+            }
+            if (modified) {
+                XContentBuilder builder = XContentFactory.contentBuilder(body.v1());
+                builder.map(body.v2());
+                bodyContent = builder.bytes();
+            }
+        }
+        RestSearchAction.parseSearchRequest(internalRequest.source(), indicesQueriesRegistry, request,
+                parseFieldMatcher, bodyContent);
+
+        String conflicts = request.param("conflicts");
+        if (conflicts != null) {
+            internalRequest.conflicts(conflicts);
+        }
+        parseCommon(internalRequest, request);
+
+        // TODO allow the user to modify the batch size? Or pick something better than just a default.
```

Comment:
Its actually still a TODO! I don't really know what to do with it so I figured it could wait until another PR.